File: app/src/main/java/com/shop440/SearchActivity.java
package com.shop440;

import android.content.Intent;
import android.graphics.Color;
import android.os.Bundle;
import android.support.v4.view.MenuItemCompat;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.SearchView;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.EditText;
import android.widget.ProgressBar;

import com.shop440.Adapters.CategorySearchAdapter;
import com.shop440.Models.CategoryModel;

import java.util.ArrayList;

public class SearchActivity extends AppCompatActivity {
    private ArrayList<CategoryModel> model;
    private RecyclerView recyclerView;
    private LinearLayoutManager linearLayoutManager;
    private CategorySearchAdapter categoryAdapter;
    private ProgressBar bar;
    private MenuItem searchItem;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_search);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        model = new ArrayList<>();
        categoryAdapter = new CategorySearchAdapter(this, model);
        recyclerView = (RecyclerView) findViewById(R.id.recyclerView);
        bar = (ProgressBar) findViewById(R.id.progressBar);
        linearLayoutManager = new LinearLayoutManager(this);
        recyclerView.setLayoutManager(linearLayoutManager);
        recyclerView.setHasFixedSize(true);
        recyclerView.setAdapter(categoryAdapter);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.search, menu);
        //MenuItem searchItem = menu.findItem(R.id.search);
        searchItem = menu.findItem(R.id.action_search);
        final SearchView searchView = (SearchView) MenuItemCompat.getActionView(searchItem);
        int searchEditId = android.support.v7.appcompat.R.id.search_src_text;
        searchItem.expandActionView();
        EditText et = (EditText) searchView.findViewById(searchEditId);
        et.setTextColor(Color.BLACK);
        et.setHintTextColor(Color.GRAY);
        et.requestFocus();
        searchView.setIconified(false);
        searchView.setIconifiedByDefault(true);
        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
            @Override
            public boolean onQueryTextSubmit(String query) {
                Intent i = new Intent(SearchActivity.this, SearchResultActivity.class);
                i.putExtra("query", query);
                i.putExtra("isSearch", true);
                startActivity(i);
                return true;
            }

            @Override
            public boolean onQueryTextChange(String newText) {
                return false;
            }
        });

        return true;
    }

}

File: app/src/main/java/com/shop440/search/SearchActivity.kt
package com.shop440.search

import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.view.MenuItemCompat
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.SearchView
import android.view.*
import android.widget.EditText
import android.widget.ProgressBar
import com.shop440.R
import com.shop440.adapters.CategorySearchAdapter
import com.shop440.dao.models.CategoryModel
import kotlinx.android.synthetic.main.activity_search.*
import java.util.*

class SearchActivity : Fragment() {
    private var model: ArrayList<CategoryModel>? = null
    private var linearLayoutManager: LinearLayoutManager? = null
    private var categoryAdapter: CategorySearchAdapter? = null
    private var bar: ProgressBar? = null
    private var searchItem: MenuItem? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_search)


    }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater?.inflate(R.layout.activity_search, container, false)
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        model = ArrayList()
        categoryAdapter = CategorySearchAdapter(this, model)
        bar = R.id.progressBar
        linearLayoutManager = LinearLayoutManager(this)
        recyclerView!!.layoutManager = linearLayoutManager
        recyclerView!!.setHasFixedSize(true)
        recyclerView!!.adapter = categoryAdapter
    }
}
